samples/bidirectional-chat/content/index.html

```diff
@@ -0,0 +1,239 @@
+
+            if (messageText) {
+              connection.invoke("sendToUser", user, messageText);
+            }
+          },
+          sendToConnection: function (connectionId) {
+            const messageText = prompt('Send private message to connection ' + connectionId);
+
+            if (messageText) {
+              connection.invoke("sendToConnection", connectionId, messageText);
+            }
+          },
+          addConnectionToGroup: function(connectionId) {
+            confirm('Add connection ' + connectionId + ' to group: ' + this.defaultgroup);
+            connection.invoke("joinGroup", connectionId, this.defaultgroup);
+          },
+          addUserToGroup: function (user) {
+            r = confirm('Add user ' + user + ' to group: ' + this.defaultgroup);
+            connection.invoke("joinUserToGroup", user, this.defaultgroup);
+          },
+          removeConnectionIdFromGroup: function(connectionId) {
+            confirm('Remove connection ' + connectionId + ' from group: ' + this.defaultgroup);
+            connection.invoke("leaveGroup", connectionId, this.defaultgroup);
+          },
+          removeUserFromGroup: function(user) {
+            confirm('Remove user ' + user + ' from group: ' + this.defaultgroup);
+            connection.invoke("leaveUserFromGroup", user, this.defaultgroup);
+          }
+        }
+      });
+      const apiBaseUrl = prompt('Enter the Azure Function app base URL', window.apiBaseUrl);
+      data.username = prompt("Enter your username");
+      const isAdmin = confirm('Work as administrator? (only an administrator can broadcast messages)');
+      if (!data.username) {
+        alert("No username entered. Reload page and try again.");
+        throw "No username entered";
+      }
+      const connection = new signalR.HubConnectionBuilder()
+        .withUrl(apiBaseUrl + '/api', {
+          accessTokenFactory: () => {
+            return generateAccessToken(data.username)
+          }
+        })
+        .configureLogging(signalR.LogLevel.Information)
+        .build();
+      connection.on('newMessage', onNewMessage);
+      connection.on('newConnection', onNewConnection)
+      connection.onclose(() => console.log('disconnected'));
+      console.log('connecting...');
+      connection.start()
+        .then(() => {
+          data.ready = true;
+          console.log('connected!');
+        })
+        .catch(console.error);
+      function getAxiosConfig() {
+        const config = {
+          headers: {
+            'x-ms-signalr-user-id':  data.username,
+            'Authorization': 'Bearer ' + generateAccessToken(data.username)
+          }
+        };
+        return config;
+      }
+      let counter = 0;
+      function onNewMessage(message) {
+        message.id = counter++; // vue transitions need an id
+        data.messages.unshift(message);
+      };
+      function onNewConnection(message) {
+        data.myConnectionId = message.ConnectionId;
```

Comment:
Do we want to do a camel case conversion for class members? (ConnectionId -> connectionId) I think SignalR by default does this conversion.